Update HT2.asl

ero-qt opened the pull request:
test before merging

ero-qt committed:
Update HT2.asl

Kiraerong commented:
its much nicer looking. unfortunately it doesnt split after reaching the percentages

Kiraerong commented:
i just retried the code and it seems to work. for some reason it wasnt working on one device

Kiraerong commented:
hey, so i discovered why it was bugged earlier. it wont split sometimes because the game can jump more than 1%. so it wont split at exactly at the 5% intervals

ero-qt committed:
Addresses issues

Kiraerong committed:
Update HT2.asl

everything works except 51% split. just a small add, and seemed to work using the CE

ero-qt committed:
simplify

Kiraerong reviewed (commented):
If a player has 49% mono, and hires a truck that takes them over 51%, itll only split one time

ero-qt committed:
Address issues

ero-qt commented:
Took me a while to get to this, sorry.

Kiraerong commented:
i also was going through stuff, ill be testing it asap

Kiraerong commented:
so i assume its skipping either the 51% mark or the 50% mark, it only splits once if a player happens to go from 49% to 51%

ero-qt commented:
> so i assume its skipping either the 51% mark or the 50% mark, it only splits once if a player happens to go from 49% to 51%

Is that not expected behavior? Do you really want 2 splits to happen at the same exact instance when both 50% and 51% are checked in the settings? That doesn't really seem very useful to me.

Kiraerong commented:
yes, i mean its not very useful; the biggest difference is when you do 100% vs 51% runs, where 51% isnt going to matter since having the monopoly isnt going to be the end condition. it would be nice to have them both split incase someone wants both

ero-qt commented:
And how does the code interfere with that? You can split on 50% just fine. If you're doing 100%, and 51% isn't the ending condition, simply don't check 51%? I'm not sure what you're getting at.

Kiraerong commented:
i mean, 51% is required for monopoly runs, 100% is required for 100%. i mean, im sure we can add a comment that says to uncheck one or the other. just making it idiot proof really.

ero-qt commented:
Could we make some progress on this PR, please? Where is this stuck again?

Kiraerong commented:
i was hoping that if the player happens to reach 50% and 51% at once, that they both split at once kind of deal. instead of it splitting at either or

Commit:
Split on every possible percentage on increase

ero-qt commented:
I believe this should do it. This now loops over every full percentage point when `mono` has increased.

Kiraerong commented:
ill give it a shot when i can. sorry, my pc hasnt been having a good time
